pkg/plugins/backendplugin/manager/manager.go

```diff
@@ -244,10 +244,7 @@ func (m *manager) stop(ctx context.Context) {
 
 // CollectMetrics collects metrics from a registered backend plugin.
 func (m *manager) CollectMetrics(ctx context.Context, pluginID string) (*backend.CollectMetricsResult, error) {
-	m.pluginsMu.RLock()
-	p, registered := m.plugins[pluginID]
-	m.pluginsMu.RUnlock()
-
+	p, registered := m.Get(pluginID)
```

Comment:
Mental note that `Get` should also probably check `IsDecommissioned()` as part of the second return. Something for later

Reply:
I pushed an update to `Get` returning nil, and false if the plugin is decommissioned